### PR TITLE
Row styles field permissions

### PR DESCRIPTION
**Pull request by bryanbraun**
# READY FOR REVIEW

## Summary

Jira card: https://sparkbox.atlassian.net/browse/STN-374

This ensures that users with the "Site Manager" or "Contributor" roles can see and update the row styles field.

These permissions changes should only apply to newly provisioned sites because permissions are currently config-ignored, as shown in [`/config/default/config_ignore.settings.yml`](https://github.com/SU-HSDO/suhumsci/blob/24b64ec3468d2fbd876c54f93066c859337b6c60/config/default/config_ignore.settings.yml#L23). Existing production sites can change their production setting 

## Need Review By (Date)
5/25

## Urgency
med

## Steps to Test
**New default permissions:**
1. Import these config changes into a local site with `drush config-import`
2. Browse to both permissions locations (`/admin/users/permissions` and `/admin/structure/paragraphs_type/hs_row/fields/paragraph.hs_row.field_paragraph_style`), and confirm that site managers and contributors now have all field permissions for the `field_paragraph_style` field.
    - Screen Shot 2020-05-20 at 6 05 41 PM
    - Screen Shot 2020-05-20 at 6 03 56 PM
3. Use masquerade to log in as a user with either the "Site Manager" or "Contributor" role, have them create a new basic page with a row, and confirm that they can see the Paragraph style field.
    - Screen Shot 2020-05-20 at 6 02 39 PM
4. Confirm that these settings are config-ignored on the line that specifies `'user.role.*:permissions'`: [`/config/default/config_ignore.settings.yml`](https://github.com/SU-HSDO/suhumsci/blob/24b64ec3468d2fbd876c54f93066c859337b6c60/config/default/config_ignore.settings.yml#L23).

**Permissions changing in production**
1. Login to the sandbox site: https://sparkbox-sandbox-prod.stanford.edu/user/login
2. Browse to https://sparkbox-sandbox-prod.stanford.edu/admin/structure/paragraphs_type/hs_row/fields/paragraph.hs_row.field_paragraph_style and confirm that:
    1. The field permissions are NOT checked for the "Site Manager" or "Contributor" roles.
    2. These field permissions are locked on this screen (they cannot be changed in prod).
3. Browse to https://sparkbox-sandbox-prod.stanford.edu/admin/users/permissions, search for the five `field_paragraph_style` permissions, check them for the "Site Manager" and "Contributor" roles, and then save the settings.
4. Browse back to https://sparkbox-sandbox-prod.stanford.edu/admin/structure/paragraphs_type/hs_row/fields/paragraph.hs_row.field_paragraph_style and confirm that the permissions have been updated.
5. Use masquerade to log in as a user with either the "Site Manager" or "Contributor" role, then create a new basic page with a row, and confirm that you can see the Paragraph style field.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

**Comment by bryanbraun** [comment]
This has been rebased and should be ready for review. 👍

File: tests/behat/features/install/users/RoleContributor.feature [hunk 1]
```diff
@@ -6,7 +6,7 @@ Feature: Install State
   @api @safe
   Scenario: Test default permissions.
     Given I run drush "cget user.role.contributor permissions"
-    Then the role "contributor" should have 82 permissions
+    Then the role "contributor" should have 87 permissions
```
Comment (silverli) [inline tests/behat/features/install/users/RoleContributor.feature:9]: cool to see these behat tests!!

File: tests/behat/features/install/users/RoleContributor.feature [hunk 2]
```diff
@@ -85,6 +89,7 @@ Feature: Install State
     And drush output should contain "view hs_person revisions"
     And drush output should contain "view hs_publications revisions"
     And drush output should contain "view hs_research revisions"
+    And drush output should contain "view own field_paragraph_style"
```
Comment (silverli) [inline tests/behat/features/install/users/RoleContributor.feature:92]: it's interesting that you can distinguish between "your own" field and "anybody's" field

**Comment by silverli** [comment]
@bryanbraun looking good! Here's some screenshots of what I validated locally:

Screen Shot 2020-05-21 at 1 24 21 PM
Screen Shot 2020-05-21 at 1 24 56 PM
Screen Shot 2020-05-21 at 1 30 41 PM

**Comment by silverli** [comment]
@bryanbraun for The Sandbox site on prod, I was able to verify that initially, the field permissions were not checked for Site Manager and Contributor, but I was able to change that, and then see that those changes were saved, and also applied in Paragraph Types -> Row -> Paragraph Style -> Field permissions.


Here is a screenshot to show I was able to create a row with a paragraph style masquerading as both Contributor and Site manager:

Screen Shot 2020-05-21 at 1 54 10 PM

**Review by silverli — APPROVED** [review]
There are two more permissions you have to add (we learned about in standup) but tests all pass, and there is no compiled CSS and everything works! I'll approve and then recheck when you commit the two other changes for user permissions on the paragraph field style

File: tests/behat/features/install/users/RoleAnonymousUser.feature [hunk 3]
```diff
@@ -6,13 +6,14 @@ Feature: Install State
   @api @safe
   Scenario: Test default permissions.
     Given I run drush "cget user.role.anonymous permissions"
-    Then the role "anonymous" should have 9 permissions
+    Then the role "anonymous" should have 10 permissions
```
Comment (silverli) [inline tests/behat/features/install/users/RoleAnonymousUser.feature:9]: 👍

File: tests/behat/features/install/users/RoleAuthenticatedUser.feature [hunk 4]
```diff
@@ -15,6 +15,7 @@ Feature: Install State
     And drush output should contain "view any event_collections entities"
     And drush output should contain "view any publications_collections entities"
     And drush output should contain "view field_hs_text_area_bg_color"
+    And drush output should contain "view field_paragraph_style"
```
Comment (silverli) [inline tests/behat/features/install/users/RoleAuthenticatedUser.feature:18]: Good that you caught these!

**Review by silverli — APPROVED** [review]
Great work! And thanks for the really thorough PR description and instructions!! It's great documentation of the issue if we ever need to refer back.

**Pull request merged**